pkg/util/fs_util.go

```diff
@@ -474,27 +491,27 @@ func CopyDir(src, dest string) ([]string, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		uid, gid := determineChownUIDGid(fi, chownUID, chownGid)
+
 		destPath := filepath.Join(dest, file)
 		if fi.IsDir() {
 			logrus.Debugf("Creating directory %s", destPath)
 
-			uid := int(fi.Sys().(*syscall.Stat_t).Uid)
-			gid := int(fi.Sys().(*syscall.Stat_t).Gid)
-
 			if err := os.MkdirAll(destPath, fi.Mode()); err != nil {
 				return nil, err
 			}
-			if err := os.Chown(destPath, uid, gid); err != nil {
+			if err := os.Chown(destPath, int(uid), int(gid)); err != nil {
 				return nil, err
 			}
 		} else if fi.Mode()&os.ModeSymlink != 0 {
 			// If file is a symlink, we want to create the same relative symlink
-			if err := CopySymlink(fullPath, destPath); err != nil {
+			if err := CopySymlink(fullPath, destPath, chownUID, chownGid); err != nil {
```

Comment:
should these be `uid` and `gid` instead of `chownUID` and `chownGid`? I think with that change `determineChownUIDGid` won't need to be called again in `CopySymlink`

Reply:
My thought here was since `CopySymlink` is exported it could be called in other places that didn't have the resolved `uid/gid` (or to allow callers to those functions to specify "keep uid the same", e.g. https://github.com/GoogleContainerTools/kaniko/pull/250/files#diff-a29f0f6c0a35589007a1c4e592f5c189R166) and that it would be less error-prone for callers if `determineChownUIDGid` was called within `CopySymlink/CopyFile`. I agree that expectation could be changed and I can change this if you'd like.

Reply:
Hi @priyawadhwa - what do you think?

Reply:
That makes sense -- we can keep the `determineChownUIDGid` in both places, but let's still pass in `uid` and `gid` in this function since it's a bit cleaner

Reply:
okay I can do that but it would have to be `int(uid)` and `int(gid)` (since those values are uint32s but the other functions expect signed ints which `determineChownUIDGid` returns) - I wasn't sure if the type casting to signed was frowned upon or not.